### PR TITLE
[UR] Add Context query enums for atomic capabilities

### LINKED ISSUE #75: Missing context info enum values
- PI_CONTEXT_INFO_PLATFORM - Its a useful information, used by some backend to make certain decision (device->platform->name). But this can also be done by urDeviceGetInfo(UR_DEVICE_INFO_PLATFORM)->urGetPlatformInfo(PLATFORM_NAME). So we can skip this.
- PI_CONTEXT_INFO_PROPERTIES - Cant find its usage
- PI_CONTEXT_INFO_REFERENCE_COUNT  Cant find its usage too. UR has the concept of context acquire/release, but no way to query its reference count. Its inconsistent, for device which also has acq/rel, we have a UR_DEVICE_INFO_REFERENCE_COUNT. can add it to make it consistent.
- PI_CONTEXT_INFO_ATOMIC_MEMORY_[ORDER|SCOPE]_CAPABILITIES - used to populate sycl device info for order/scope capabilities, Should they be moved to device info instead of context?

### PR DESCRIPTION
**Pull request by veselypeta**
Closes #75

### CODE REVIEW

File: scripts/core/context.yml [hunk 1]
```diff
@@ -98,6 +98,14 @@ etors:
...
     - name: USM_FILL2D_SUPPORT
       desc: "[bool] to indicate if the $xEnqueueUSMFill2D entrypoint is supported."
+    - name: ATOMIC_MEMORY_ORDER_CAPABILITIES
+      desc: "[$x_memory_order_capability_flags_t] return a bit-field of atomic memory order capabilities."
+    - name: ATOMIC_MEMORY_SCOPE_CAPABILITIES
+      desc: "[x_memory_scope_capability_flags_t] return a bit-field of atomic memory scope capabilities."
+    - name: ATOMIC_FENCE_ORDER_CAPABILITIES
+      desc: "[$x_memory_order_capability_flags_t] return a bit-field of atomic memory fence order capabilities."
+    - name: ATOMIC_FENCE_SCOPE_CAPABILITIES
+      desc: "[x_memory_scope_capability_flags_t] return a bit-field of atomic memory fence scope capabilities."
```
Comment (kbenzie) [inline scripts/core/context.yml:108]: I'd expect `0` to be returned if context level atomic fences aren't supported, this is only implemented in the CUDA adapter currently I believe. Perhaps we should mention this in the descriptions?

**Review by kbenzie — CHANGES_REQUESTED** [review]

File: scripts/core/context.yml (hunk 1 above)
Comment (veselypeta) [reply scripts/core/context.yml]: 👍

**Review by kbenzie — APPROVED** [review]
LGTM

**Review by martygrant — APPROVED** [review]
LGTM

**Review by callumfare — APPROVED** [review]

**Pull request merged**